testcase.py

```diff
@@ -114,7 +115,7 @@ def __initilize_data(self):
         self.outputs_metadata = self._get_var_metadata(self.fmu, self.output_names)
         self.forecasts_metadata = self.data_manager.get_data_metadata()
         # Outputs data
-        self.y = {'time': np.array([])}
+        self.y = {'time': a.array('f',[])}
```

Comment:
By conclusions, do you mean memory efficiency or conclusions of tests (e.g. kpis, results)?  I will check memory efficiency and if source of diffs.

Reply:
I meant memory efficiency. 

But also might the kpi diffs go away if we use double precision `d` instead of float `f`?

Reply:
Thanks, good point.  I will check on both.

Reply:
I'm not actually sure what precision numpy is using, but I think if you default construct an empty numpy array it will probably be a 64 bit float.  https://stackoverflow.com/questions/61853331/which-float-precision-are-numpy-arrays-by-default.

With numpy array you can also be explicit about the precision by passing dtype parameter https://numpy.org/devdocs/reference/generated/numpy.array.html

Reply:
Updated plot with using python array with "d" (in pink) and numpy array with dtype="float32" (in gray), for bestest_hydronic with 60s step for 180 day simulation:
![Figure_1-7](https://user-images.githubusercontent.com/19480999/221967263-a99dd8fa-2f14-4f32-94c0-5bd9e0aeedd7.png)


With python array with 64-bit float, looks like the speed benefit is still there, but the total memory reduction is not, which I guess makes sense given your insights.

Also ran the unit test for bestest_hydronic_heat_pump with python array and using "d" indeed leads to a pass without the diffs.

Reply:
Sounds like python array with double precision might be a winner? I guess one question is if the use case requires double precision or not. I believe there was a famous debate within the EnergyPlus team about this. 

@amirroth

Reply:
There doesn't seem to be any discernible difference in reported results between single and double within 7 or 8 significant digits.

Reply:
I'm actually perfectly supportive either way we go. I'm mostly interested in the computational performance gain. 32 bit single precision seems perfectly adequate if you're ok with the diffs. When you're in the loop of a numerical solver where error may accumulate then I can see the case for double precision but here - meh.

Reply:
> Sounds like python array with double precision might be a winner? I guess one question is if the use case requires double precision or not. I believe there was a famous debate within the EnergyPlus team about this.
> 
> @amirroth

32-bit float gives you about 8 digits of precision and a range of 10E-38 to 10E38. My contention has always been that nothing in building energy modeling needs more than that range and certainly not more than this precision (the only thing that is known with more than 8 digits of precision in BEM is Pi). Of course, there is the "what about subtractive cancellation" peanut gallery. Yes, that is a problem in a few linear algebra algorithms, including I believe matrix inversion. But if you are really worried about that you can just convert to `double` inside those algorithms or use known alternatives that don't have these issues.

Incidentally, I am of the same opinion about 32-bit integers.

Reply:
> Does that mean the unit test is looking for more digits than 7 or 8?

The numeric comparisons use both a relative and absolute tolerance, where the absolute tolerance is 1e-3, so for larger numbers, sometimes the absolute error can trigger a failure.

Since the application here is storing data from simulation outputs and not within numerical solver algorithms, I'm inclined to use 32 bit precision for both computational performance gains and some memory use reduction, which could help with more complex emulators that produce more data down the line.

This PR should be ready to merge when tests pass and if there are no objections.

Reply:
Sounds great to me.